### PR TITLE
Add missing OCaml C runtime macros, occasional style changes as well

### PR DESCRIPTION
**Pull request by fxfactorial**
Signed the CLA as well.

I have half a mind to reimplement the command line with cmdliner, you guys are missing out on a beautiful API and auto generated man page and most of all much less code under command.

### CODE REVIEW

**Comment by mroch** [comment]
cc @int3 @jwatzman @dlreeves for the hack/ changes

**Comment by jwatzman** [comment]
Not on the Hack team anymore, @int3 and @dlreeves should take a look. I did a *very* quick skim, it looks fine to me, except perhaps the changes to `src/embedded/flowlib_elf.c` -- not clear if those `free` calls are correct, would have to look at the libelf API again (ugh).

**Comment by fxfactorial** [comment]
@jwatzman I can remove those `free` calls to play it safe, it did look like a memory leak on the face of it.

**Comment by mroch** [comment]
the linux travis builds are segfaulting :(

**Comment by fxfactorial** [comment]
@mroch Okay sorry about that, I will try building it as well and fix it.

**Comment by fxfactorial** [comment]
Okay this is odd, I don't get a segfault on my ubuntu VM, albeit using later versions libs and compilers:

http://pastebin.com/VEnq118f

Also using a `nixpkg` version of `libelf`.  I will try to reproduce the build environment exactly.

**Comment by fxfactorial** [comment]
@jwatzman @mroch @int3 @dlreeves I took away the three `free` calls in the `elf` function call and now the seg fault goes away. 

Sorry for delay.

**Comment by gabelevi** [comment]
This looks good to me too.

**Comment by mroch** [comment]
@facebook-github-bot shipit

**Comment by mroch** [comment]
thanks, let's do this

**Comment by fxfactorial** [comment]
I can't tell if this was actually merged? Should I close this PR?

**Comment by samwgoldman** [comment]
The commit message will autoclose this when it lands. Looks like there are some build errors internally that are preventing this from landing.

**Comment by fxfactorial** [comment]
@samwgoldman hmm, a little hard for me to fix them if I can't see them.

**Comment by samwgoldman** [comment]
Yeah, I agree. We'll look into it ASAP.

File: hack/heap/hh_shared.c [hunk 1]
```diff
@@ -592,7 +594,7 @@ void hh_add_dep(value ocaml_dep) {
   htable_add(deptbl, dep >> 31, dep);
```
Comment (mroch) [inline hack/heap/hh_shared.c]: ah, finally figured out what was wrong in our internal tests (and I'm still not sure why it doesn't repro in travis).

adding `CAMLparam1` here without also adding `CAMLreturn0` here and the `return` a couple lines up meant that `ocaml_dep` was never freed and it'd OOM.

I'm including that fix, no need to update. should be landing soon. thanks for your patience!

Comment (fxfactorial) [inline hack/heap/hh_shared.c]: @mroch Much appreciated, thanks! (Also found FB infer recently, awesome work guys! love all what you do)

**Pull request closed without merging**